src/types/submission.ts

```diff
@@ -35,37 +40,51 @@ export interface WebhookView {
   data: WebhookData
 }
 
+export interface ISubmissionSchema extends ISubmission, Document {
+  getWebhookView(): WebhookView | null
+}
+
 export interface IEmailSubmission extends ISubmission {
-  recipientEmails?: string[]
+  recipientEmails: string[] | []
   responseHash: string
   responseSalt: string
   hasBounced: boolean
+  encryptedContent: never
+  verifiedContent: never
+  version: never
+  attachmentMetadata: never
+  webhookResponses: never
+  getWebhookView(): WebhookView | null
 }
 
-export interface IEmailSubmissionSchema
-  extends IEmailSubmission,
-    ISubmissionSchema {}
+export type IEmailSubmissionSchema = IEmailSubmission & ISubmissionSchema
 
 export interface IEncryptedSubmission extends ISubmission {
+  recipientEmails: never
+  responseHash: never
+  responseSalt: never
+  hasBounced: never
   encryptedContent: string
   verifiedContent?: string
   version: number
   attachmentMetadata?: Map<string, string>
-  webhookResponses: IWebhookResponseSchema[]
+  webhookResponses: IWebhookResponse[] | []
+  getWebhookView(): WebhookView | null
 }
 
-export interface IWebhookResponseSchema extends Document {
+export type IEncryptedSubmissionSchema = IEncryptedSubmission &
+  ISubmissionSchema
+
+export interface IWebhookResponse {
   webhookUrl: string
   signature: string
-  errorMessage: string
-  response: {
+  errorMessage?: string
+  response?: {
     status: number
```

Comment:
Typing doesn't look right according to what I saw above. The child keys of `response` can also be `undefined` right? Due to 

```js
response: {
      status: get(response, 'status'),
      statusText: get(response, 'statusText'),
      headers: stringifySafe(get(response, 'headers', {})),
      data: stringifySafe(get(response, 'data', {})),
},
```

Reply:
fixed